src/token/erc1155/erc1155.cairo

```diff
@@ -0,0 +1,545 @@
+        #[key]
+        from: ContractAddress,
+        #[key]
+        to: ContractAddress,
+        #[key]
+        id: u256,
+        value: u256
+    }
+
+    /// Emitted when `values` are transferred from `from` to `to` for `id`.
+    #[derive(Drop, starknet::Event)]
+    struct TransferBatch {
+        #[key]
+        operator: ContractAddress,
+        #[key]
+        from: ContractAddress,
+        #[key]
+        to: ContractAddress,
+        #[key]
+        ids: Span<u256>,
+        values: Span<u256>,
+    }
+
+    /// Emitted when `account` enables or disables (`approved`) `operator` to manage
+    /// all of its assets.
+    #[derive(Drop, starknet::Event)]
+    struct ApprovalForAll {
+        #[key]
+        account: ContractAddress,
+        #[key]
+        operator: ContractAddress,
+        approved: bool
+    }
+
+    mod Errors {
+        const INVALID_TOKEN_ID: felt252 = 'ERC1155: invalid token ID';
+        const INVALID_ACCOUNT: felt252 = 'ERC1155: invalid account';
+        const INVALID_OPERATOR: felt252 = 'ERC1155: invalid operator';
+        const UNAUTHORIZED: felt252 = 'ERC1155: unauthorized caller';
+        const APPROVAL_TO_OWNER: felt252 = 'ERC1155: approval to owner';
+        const SELF_APPROVAL: felt252 = 'ERC1155: self approval';
+        const INVALID_RECEIVER: felt252 = 'ERC1155: invalid receiver';
+        const ALREADY_MINTED: felt252 = 'ERC1155: token already minted';
+        const WRONG_SENDER: felt252 = 'ERC1155: wrong sender';
+        const SAFE_MINT_FAILED: felt252 = 'ERC1155: safe mint failed';
+        const SAFE_TRANSFER_FAILED: felt252 = 'ERC1155: safe transfer failed';
+        const INVALID_ARRAY_LEN: felt252 = 'ERC1155: no equal array length';
+        const INVALID_ARRAY_LENGTH: felt252 = 'ERC1155: invalid array length';
+        const INSUFFICIENT_BALANCE: felt252 = 'ERC1155: insufficient balance';
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(ERC1155Impl)]
+    impl ERC1155<
+        TContractState,
+        +HasComponent<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of interface::IERC1155<ComponentState<TContractState>> {
+        /// Returns the number of NFTs owned by `account` from a specific token ID.
+        fn balance_of(
+            self: @ComponentState<TContractState>, account: ContractAddress, token_id: u256
+        ) -> u256 {
+            assert(account.is_non_zero(), Errors::INVALID_ACCOUNT);
+            self.ERC1155_balances.read((token_id, account))
+        }
+
+
+        /// Returns a span of u256 values representing the batch balances of the
+        /// accounts for the specified token IDs.
+        ///
+        /// Requirements:
+        ///
+        /// - `token_ids` and `accounts` must have the same length.
+        fn balance_of_batch(
+            self: @ComponentState<TContractState>,
+            accounts: Span<ContractAddress>,
+            token_ids: Span<u256>
```

Comment:
Just easier to follow imo, from the fact that param names doesn't affect the interface. The standard also uses owners instead of accounts, and our solidity library version uses accounts, which is different as well. Do you think we should stick with the param names from the standard?

Reply:
I am agreed with @ericnordelo , I made the choice when I implemented the erc-1155. As the standard ask for `id` in interface, I think the use of `id` in the code lack of clarity in reading the code to understand it. Especially since the term `id` is a legacy of the ERC-1155, we can keep it only for the interface IMO

Reply:
i find `ids` equally easy to follow, given the context of a token contract. same with `uri`, it's just uri. I don't have strong opinions here but I usually lean towards respecting the spec as much as possible, unless there's a clear advantage in not doing so.